src/test.js

```diff
@@ -131,57 +127,73 @@ class Test {
             throw 'screenDefaults are required!';
         }
         this.homeScreen = new HomeScreen(this.language, this.screenDefaults);
+        this.loopUtilities = new LoopUtilities(this);
```

Comment:
I'm confused. I see here you are creating a new instance of `LoopUtilities` as `loopUtilities`, but throughout the rest of this PR, I see you are creating a new instances everywhere. That is `this.LoopUtilities()...`. I think one created instance of `test.loopUtilities` is the better way to go here.

[nit] Also, this code is all "Loop" related, so perhaps just `Utilities` and `test.utilities`?